src/formatter/javascript.ts

```diff
@@ -73,7 +80,25 @@ export class SyntheticsGenerator extends JavaScriptLanguageGenerator {
     this.previousContext = undefined;
   }
 
-  generateAction(actionInContext: ActionInContext) {
+  /**
+   * Generate code for an action.
+   * @param actionInContext The action to create code for.
+   * @param ignoreStepDefaults When true, the function will not create steps between
+   * navigation events as it normally would.
+   * @param customStepsOverride When true, the function will not throw an error if the
+   * called while the generator is already recording a step.
+   * @returns the strings generated for the action.
+   */
+  generateAction(
+    actionInContext: ActionInContext,
+    ignoreStepDefaults = false,
+    customStepsOverride = false
```

Comment:
I couldn't understand what this parameter does and why we need this one - could you tell me more about it please?

Reply:
I'm happy to accept suggestions for a more appropriate name here. Perhaps we can call it `ignoreNewStepLogic`?